docs/jsonapi-spec.textile

```diff
@@ -52,6 +52,104 @@ bc(sample). SELECT sample_usage FROM cql;
 
 Samples of requests, operations, and clauses will be presented and encoded as a JSON document. This is illustrative of a suggested encoding as described in the "Preamble":#preamble   
 
+h2(#messaging). Request and Response Messages
+
+The format of Request and Response messages is included in this document so that the format invoking a command, and the structrure of it's response can be understood. 
+
+h3(#messagingRequest). Request Messages
+
+Request messages contain only a single command. Batch operations such as bulk loading are implementing using commands that support the inclusion of multiple documents in the command. The request message is a single JSON document (a JSON Object) that represents the command to run.
+
+p. _Syntax:_
+
+bc(syntax).. 
+<request-message> ::=  <command-name> <command>
+
+p. @<command-name>@ and @<command>@ are defined later in this document document. 
+
+p. _Sample:_
+
+bc(sample).. 
+{"find" : {
+    "filter" : {"name": "aaron"},
+    "projection" : {"name": 1, :"age" : 1}
+    } 
+} 
+
+p. Top level names in the request message other than a command name are ignored, except where specified in this document.
+
+h3(#messagingResponse). Response Messages
+
+A single JSON format is used for all response messages, though different elements may be present depending on the command. 
+
+p. _Syntax:_
+
+bc(syntax).. 
+<response-message> ::=  (errors (<error>)+)?,    
+                        (status (<command-status>)+)?, 
+                        (data <response-data>)?,
+
+<<error> ::= message <ascii-string>, 
+             errorCode <ascii-string>,
```

Comment:
Not sure about the error code currently, but I think we don't have it implemented at the moment.

Reply:
I got the errorCode from examples provided by @tatu-at-datastax

Reply:
Yes I think it is returned and integration tests verify it, along with `message`.